src/selector.rs

```diff
@@ -0,0 +1,24 @@
+//! Abstract Faiss ID selector
+use faiss_sys::*;
+
+/// Abstraction over IDSelectorRange and IDSelectorBatch
+#[derive(Debug)]
+pub struct IdSelector {
+    pub inner: *mut FaissIDSelector,
+}
+
+impl IdSelector {
+    /// Create new range selector
+    pub fn range(min: idx_t, max: idx_t) -> IdSelector {
```

Comment:
Please use `Idx` for the vector idx type instead. It is currently an alias to `idx_t`, but as a higher level abstraction, this might change in the future.

In addition, since this constructor function returns an error code, this one should also return a `Result` (from the `error` module).

```suggestion
    pub fn range(min: Idx, max: Idx) -> Result<Self> {
```